ovh/ovh.go

```diff
@@ -114,6 +127,21 @@ func NewDefaultClient() (*Client, error) {
 	return NewClient("", "", "", "")
 }
 
+func NewOAuth2Client(endpoint, clientID, clientSecret string) (*Client, error) {
+	client := Client{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		Client:       &http.Client{},
```

Comment:
http.Client{Timeout: DefaultTimeout}

By the way, why not using the http.DefaultClient (that might be customized by the user of this package) and let someone who wants a new client customize it if needed (since it is a public attribute) ?

Reply:
`Timeout` is already injected at every requests in the client.
https://github.com/ovh/go-ovh/blob/133b8b82502e33b78bbd2e8cb3271e27f72181fb/ovh/ovh.go#L357
I'm not changing the previous behavior here.

Regarding the DefaultClient suggestion, it could be great, let's do this in another PR.